src/main/algorithms/search/algorithms/depthFirstSearch.ts

```diff
@@ -1,138 +1,154 @@
         const pb = new ProtocolBuilder<SearchSimulationStep>()
         pb.setStepCloner((step: SearchSimulationStep) => cloneSearchSimulationStep(step))
         const startItemCoords = start.getValue().item.data().coords
         const endItemCoords = end.getValue().item.data().coords
 
-        const visitedVertices: Vertex<VertexValue>[] = [start]
-        const edgeStack: Edge<VertexValue, EdgeValue>[] = graph.getEdges().filter(e => e.getFrom() === start)
+        const vertexStack: Vertex<VertexDepthFirstSearchValue>[] = [start]
 
-        const path: GraphFormItem[] = [start.getValue().item]
-        this.createStep(graph, grid, start, startItemCoords, endItemCoords, visitedVertices, pb)
+        let highlightedGrid = this.createStep(graph.getVertices(), grid)
+        pb.step({ grid: highlightedGrid, start: highlightedGrid[startItemCoords.y][startItemCoords.x], end: highlightedGrid[endItemCoords.y][endItemCoords.x] })
 
-        while (edgeStack.length > 0) {
-            const currentEdge: Edge<VertexValue, EdgeValue> = edgeStack.pop() as Edge<VertexValue, EdgeValue>
-            const current: Vertex<VertexValue> = currentEdge.getTo()
+        while (vertexStack.length > 0) {
+            const current: Vertex<VertexDepthFirstSearchValue> = vertexStack.pop() as Vertex<VertexDepthFirstSearchValue>
+            if (current.getValue().visited??false) continue
 
-            if (visitedVertices.includes(current)) continue
+            current.getValue().visited = true
 
-            path.push(...currentEdge.getValue().items)
-            this.createEdgeSteps(graph, grid, currentEdge, startItemCoords, endItemCoords, visitedVertices, pb)
-            visitedVertices.push(current)
-            this.createStep(graph, grid, current, startItemCoords, endItemCoords, visitedVertices, pb)
-            path.push(current.getValue().item)
+            highlightedGrid = this.createStep(graph.getVertices(), grid)
+            pb.step({ grid: highlightedGrid, start: highlightedGrid[startItemCoords.y][startItemCoords.x], end: highlightedGrid[endItemCoords.y][endItemCoords.x] })
 
             if (current === end) break
 
             const edgesToNeighbours = graph.getEdges().filter(e => e.getFrom() === current)
             for (const edge of edgesToNeighbours) {
                 const to = edge.getTo()
-                if (visitedVertices.includes(to)) {
+                if ((to.getValue().visited??false) || vertexStack.includes(to)) {
                     continue
                 }
-                edgeStack.push(edge)
+                to.getValue().predecessor = edge
+                vertexStack.push(to)
+
+                if (edge.getWeight() > 1) {
+                    this.visualiseEdgeSteps(edge, startItemCoords, endItemCoords, highlightedGrid, pb)
+                    highlightedGrid = this.createStep(graph.getVertices(), grid)
```

Comment:
Should create a step calling pb.step (same behavior as this.createPathStep). If this behavior is not wanted, the method should be named different.